src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -340,6 +379,70 @@ public void validate(FailureCollector collector) {
+            failureCollector.addFailure(String.format("Valid content type for avro is %s",
+                                                      CONTENT_TYPE_APPLICATION_AVRO), null)
+              .withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_JSON:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_JSON)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for json are %s, %s", CONTENT_TYPE_APPLICATION_JSON,
+              CONTENT_TYPE_TEXT_PLAIN), null
+            ).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_CSV:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for csv are %s, %s, %s", CONTENT_TYPE_APPLICATION_CSV,
+              CONTENT_TYPE_TEXT_PLAIN,
+              CONTENT_TYPE_TEXT_CSV), null
+            ).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_DELIMITED:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_APPLICATION_CSV)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_TSV)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for delimited are %s, %s, %s, %s", CONTENT_TYPE_TEXT_PLAIN,
+              CONTENT_TYPE_TEXT_CSV, CONTENT_TYPE_APPLICATION_CSV, CONTENT_TYPE_TEXT_TSV), null
+            ).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_PARQUET:
+          if (!contentType.equalsIgnoreCase(DEFAULT_CONTENT_TYPE)) {
+            failureCollector.addFailure(String.format("Valid content type for parquet is %s",
+                                                      DEFAULT_CONTENT_TYPE),
+                                        null).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_ORC:
+          if (!contentType.equalsIgnoreCase(DEFAULT_CONTENT_TYPE)) {
+            failureCollector.addFailure(String.format("Valid content type for orc is %s", DEFAULT_CONTENT_TYPE),
+                                        null).withConfigProperty(NAME_CONTENT_TYPE);
+          }
+          break;
+        case FORMAT_TSV:
+          if (!contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_PLAIN)
+            && !contentType.equalsIgnoreCase(CONTENT_TYPE_TEXT_TSV)) {
+            failureCollector.addFailure(String.format(
+              "Valid content types for tsv are %s, %s", CONTENT_TYPE_TEXT_TSV, CONTENT_TYPE_TEXT_PLAIN), null
```

Comment:
is `text/plain` supported for this? If so, we should add that to plugin documentation.

Reply:
PR Updated.